### PR TITLE
Meera/add amr

### PR DESCRIPTION
**Pull request by Meera-Srinivasan**
# Description

 Added code to run AMR detection as part of the host filtering step, dump results in a database, and visualize on a separate page in table form. 
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1.  

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?

- [ ] Login Screen
- [ ] Single Upload Page
- [ ] Batch Upload Page
- [ ] All Projects Page
- [ ] Sample Details Page
- [ ] Report Page

Specific components 

* 

# Testing Script:

###Login Page
* Successfully login to the application and get taken to the project page

###Single Upload
* Using an AWS path that doesn't exist throws an error
* Succesfully uploaded files from a folder (use test host)
* Uploading a duplicate file throws an error
* Pipeline gets initiated successfully 

###Batch Upload
* Using an AWS path that doesn't exist throws an error
* Succesfully uploaded files from a folder (use test host)
* Navigate to batch upload settings page after upload 
* Host selection from previous page still selected 
* Uploading a duplicate file throws a warning
* Pipeline gets initiated successfully 

###All Project Page
* Switch between projects 
* Search for a sample
* Navigate through different pages of samples
* Select Upload new sample button
* Select sample to see generated report

###Sample Details Page
* Information is consistent from Project Page
* Notes are editable and save successfully 
* Can navigate to report page 

###Report Page
* Data loads succesfully 
* Collapse and Uncollapse button works
* Clicking on the taxonomy name gives you total reads for that taxon
* Category filter works
* Genus Search works
* Filters work as expected
* Disabling Filters works
* Filtering by column works 

# Checklist:

- [x ] I have run through the testing script to make sure current functionality is unchanged
- [in progress ] I have done relevant tests that prove my fix is effective or that my feature works
- [in progress ] I have spent time testing out edge cases for my feature
- [ ] I have updated the test script or pull request template if necessary
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/controllers/samples_controller.rb [hunk 1]
```diff
@@ -188,6 +188,11 @@ def show
     end
   end
 
+  def show_amr_counts
+    @sample = samples_scope.find(params[:id])
```
Comment (cdebourcy) [inline app/controllers/samples_controller.rb]: Since you listed show_amr_counts as a read_action (which makes sense), I think you should replace `samples_scope` by `current_power.samples`. As you can see in the definition of samples_scope, it selects just the "updatable" samples (tighter access control appropriate for edit actions), but what we want is instead all the "viewable" samples, which are accessible as `current_power.samples`. If you inspect `app/models/power.rb`, you'll see that the latter expression through the `viewable` method implemented in the Sample model.

File: app/lib/dags/host_filter.json.erb [hunk 2]
```diff
@@ -5,6 +5,7 @@
     <% if @attribute_dict[:fastq2] %>
       "fastqs": ["<%= @attribute_dict[:fastq1] %>", "<%= @attribute_dict[:fastq2] %>"],
       "star_out": ["unmapped1.<%= @attribute_dict[:file_type] %>", "unmapped2.<%= @attribute_dict[:file_type] %>"],
+      "srst2_out": ["output.log", "output__genes__ARGannot_r2__results.txt", "output__fullgenes__ARGannot_r2__results.txt"],
```
Comment (cdebourcy) [inline app/lib/dags/host_filter.json.erb]: Since this line is the same in both the `if` and the `else` clause, you should take it out of the conditional to make the code more maintainable. Otherwise, any change to srst2 output files will require 2 lines of code to be modified instead of 1.

File: app/models/pipeline_run.rb [hunk 3]
```diff
@@ -26,9 +28,11 @@ class PipelineRun < ApplicationRecord
   PIPELINE_VERSION_FILE = "pipeline_version.txt".freeze
   STATS_JSON_NAME = "stats.json".freeze
   ERCC_OUTPUT_NAME = 'reads_per_gene.star.tab'.freeze
+  AMR_OUTPUT_NAME = 'output__fullgenes__ARGannot_r2__results.txt'.freeze
```
Comment (cdebourcy) [inline app/models/pipeline_run.rb]: Since you define this constant here, you should reuse it in host_filter.json.erb, rather than typing it out again. That way, the consistency of the filename you need in pipeline_run.rb and the filename that gets generated in idseq-dag is always guaranteed. It's more maintainable if you only need to modify the filename in 1 constant, rather than 2 places of the codebase.

File: app/models/pipeline_run.rb [hunk 4]
```diff
@@ -26,9 +28,11 @@ class PipelineRun < ApplicationRecord
...
   TAXID_BYTERANGE_JSON_NAME = 'taxid_locations_combined.json'.freeze
   ASSEMBLY_STATUSFILE = 'job-complete'.freeze
   LOCAL_JSON_PATH = '/app/tmp/results_json'.freeze
+  LOCAL_TXT_PATH = '/app/tmp/results_txt'.freeze
```
Comment (cdebourcy) [inline app/models/pipeline_run.rb]: I know you just wrote something analogous to what already existed (LOCAL_JSON_PATH), but I actually think it would be better to name it something more descriptive, such as `LOCAL_AMR_PATH = '/app/tmp/amr_results'.freeze`.

File: app/models/pipeline_run.rb [hunk 5]
```diff
@@ -276,6 +281,30 @@ def db_load_ercc_counts
...
   end
 
+  def download_amr_results
+    amr_s3_path = "#{host_filter_output_s3_path}/#{AMR_OUTPUT_NAME}"
+    amr_results = PipelineRun.download_file(amr_s3_path, local_txt_path)
+    amr_results
+  end
+
+  def db_load_amr_counts
+    amr_results = download_amr_results
```
Comment (cdebourcy) [inline app/models/pipeline_run.rb]: I'd rename `amr_results` to indicate that it is a file path.

Also, the fact that the code for downloading this file is spread over 2 different functions makes it a bit confusing to see what's going on. I'd almost just write 
```
amr_results_file = PipelineRun.download_file("#{host_filter_output_s3_path}/#{AMR_OUTPUT_NAME}",
  "#{LOCAL_AMR_PATH}/#{id}")
```
and remove the functions `local_txt_path` and `download_amr_results`. That way it is obvious what we're doing: download the AMR output from S3 to a local folder unique to the pipeline_run id. [If `local_txt_path` were to stay, it should definitely be renamed to something that indicates its purpose, which is to make sure we have a unique folder for each pipeline_run id.]

Even better, write:
```
amr_results_file = PipelineRun.download_file(s3_file_for("amr_counts"), "#{LOCAL_AMR_PATH}/#{id}")
```
so that you don't have to write `"#{host_filter_output_s3_path}/#{AMR_OUTPUT_NAME}"` out twice (again, this makes it more maintainable).

File: app/views/samples/show_amr_counts.html.erb [hunk 6]
```diff
@@ -0,0 +1,45 @@
+<p>
```
Comment (cdebourcy) [inline app/views/samples/show_amr_counts.html.erb]: We'll work more on this display later -- good enough for now

File: app/views/samples/show_amr_counts.html.erb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            <td> <%= amr_count.allele  %></td>
+            <td><%= amr_count.coverage %> &nbsp;</td>
+            <td><%= amr_count.depth %> &nbsp;</td>
+          </tr>
+        <% end %>
+      </tbody>
+</table>
+
+<p>
+<%= link_to 'See all samples', samples_path %> |
```
Comment (cdebourcy) [inline app/views/samples/show_amr_counts.html.erb]: What happens when you click on this? Not sure it is a page we'd want the user to see. Could change so that it takes you back to this specific sample's report page?

**Comment by Meera-Srinivasan** [comment]
Thanks for the comments @cdebourcy ! Will change + add.

File: app/models/pipeline_run.rb (hunk 4 above)
Comment (Meera-Srinivasan) [reply app/models/pipeline_run.rb]: sounds good

File: db/schema.rb [hunk 8]
```diff
@@ -25,6 +25,16 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "amr_counts", force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8" do |t|
```
Comment (cdebourcy) [inline db/schema.rb]: make sure you add a unique index on [pipeline_run_id, allele] so as to avoid your earlier issue of having duplicate entries. (See https://github.com/chanzuckerberg/idseq-web/blob/b8e5edde16c9d4d9bfca828870699ea30cafcb8e/db/schema.rb#L331 -- do it using an add_index migration)

File: app/models/pipeline_run.rb (hunk 5 above)
Comment (Meera-Srinivasan) [reply app/models/pipeline_run.rb]: Resolved -- from a testing perspective, though, I thought perhaps two functions would be easier? i.e. can test downloading separately that way. But this definitely makes sense from a readability perspective and have made this change!

File: app/controllers/samples_controller.rb (hunk 1 above)
Comment (Meera-Srinivasan) [reply app/controllers/samples_controller.rb]: Sounds good. I'm going to work on rewriting the display end for amr so hopefully I can just get rid of this soon -- will keep it in mind as I'm making my next set of display related changes.

File: app/models/pipeline_run.rb (hunk 3 above)
Comment (Meera-Srinivasan) [reply app/models/pipeline_run.rb]: for sure, I don't think the constant was recognized when I moved it to the json.erb? Did <%= @AMR_OUTPUT_NAME%> (now this constant is called some thing else, but still)

Comment (cdebourcy) [reply app/models/pipeline_run.rb]: You can access it from outside the class using `PipelineRun:: AMR_OUTPUT_NAME`

File: app/lib/dags/host_filter.json.erb (hunk 2 above)
Comment (Meera-Srinivasan) [reply app/lib/dags/host_filter.json.erb]: for sure, TODO: remove from here

File: app/controllers/samples_controller.rb (hunk 1 above)
Comment (cdebourcy) [reply app/controllers/samples_controller.rb]: Actually never mind. You should remove that line entirely, because `set_sample` is listed as a `before_action` for all `READ_ACTIONS`. So `@sample = samples_scope.find(params[:id])` happens automatically when show_amr_counts is called (just before)

File: app/lib/dags/experimental.json.erb [hunk 9]
```diff
@@ -0,0 +1,32 @@
+{
+  "output_dir_s3":
+    "s3://<%= @attribute_dict[:bucket] %>/samples/<%= @project_id %>/<%= @sample_id %>/expt",
+  "targets": {
+    <% if @attribute_dict[:fastq2] %>
+      "fastqs": ["<%= @attribute_dict[:fastq1] %>", "<%= @attribute_dict[:fastq2] %>"],
+      "srst2_out": ["output.log", "output__genes__ARGannot_r2__results.txt", "output__fullgenes__ARGannot_r2__results.txt", "amr_processed_results.csv", "amr_summary_results.csv"]
```
Comment (cdebourcy) [inline app/lib/dags/experimental.json.erb]: same comment about deduplicating by moving the `srst2_out` line after the if-else block

File: app/models/pipeline_run.rb [hunk 10]
```diff
@@ -775,6 +844,13 @@ def subsample_suffix
 
   delegate :sample_output_s3_path, to: :sample
 
+  def expt_output_s3_path
```
Comment (cdebourcy) [inline app/models/pipeline_run.rb]: If it works and you have time, this would be a good opportunity to dedup and refactor some code.  `alignment_output_s3_path`, `postprocess_output_s3_path` and now `expt_output_s3_path` all contain essentially the same code. So you could make a helper function to which you would pass `sample.sample_expt_s3_path` as an argument

File: app/models/pipeline_run_stage.rb [hunk 11]
```diff
@@ -269,8 +270,28 @@ def postprocess_command
...
+    batch_command = [install_pipeline, upload_version, dag_commands].join("; ")
+
+    # Dispatch job
+    memory = sample.sample_memory.present? ? sample.sample_memory : Sample::DEFAULT_MEMORY_IN_MB
+    aegea_batch_submit_command(batch_command, memory)
+  end
+
   def assembly_command
     # TODO: Change the following to DAG
+    # TODO: DOES this have anything to do with experimental stage
```
Comment (cdebourcy) [inline app/models/pipeline_run_stage.rb]: no, it's deprecated and being removed

File: db/migrate/20180808030435_add_drug_gene_hits_to_amr_counts.rb [hunk 12]
```diff
@@ -0,0 +1,5 @@
+class AddDrugGeneHitsToAmrCounts < ActiveRecord::Migration[5.1]
+  def change
+    add_column :amr_counts, :drug_gene_hits, :integer
```
Comment (cdebourcy) [inline db/migrate/20180808030435_add_drug_gene_hits_to_amr_counts.rb]: this is good for now until you get everything working. But before merging to master, we should consolidate all of these migrations into a single one. Probably inside create_amr_counts. You can rollback your local migrations that have already been applied using `rails db:rollback` multiple times. Then edit the file `db/migrate/20180726220853_create_amr_counts.rb` to add these additional columns. Then `rails db:migrate` again.
You should know that there was no need to do `rails generate migration` multiple times -- you could have done it ones and then edited the migration file to add more `add_column` statements

File: app/models/pipeline_run_stage.rb (hunk 11 above)
Comment (Meera-Srinivasan) [reply app/models/pipeline_run_stage.rb]: sounds good

File: db/migrate/20180808030435_add_drug_gene_hits_to_amr_counts.rb (hunk 12 above)
Comment (Meera-Srinivasan) [reply db/migrate/20180808030435_add_drug_gene_hits_to_amr_counts.rb]: Great, thanks for that suggestion !

File: app/models/pipeline_run.rb (hunk 3 above)
Comment (Meera-Srinivasan) [reply app/models/pipeline_run.rb]: Hmm I got a JSON format error when doing it that way -- will mark this as a TODO and figure out how to get it in there.

File: app/models/pipeline_run.rb [hunk 13]
```diff
@@ -276,6 +291,48 @@ def db_load_ercc_counts
     update(total_ercc_reads: ercc_counts_array.map { |entry| entry[:count] }.sum)
   end
 
+  def db_load_amr_counts
+    # test files: "s3://idseq-database/test/AMR/amr_processed_results.csv", "s3://idseq-database/test/AMR/amr_summary_results.csv"
+    amr_results = PipelineRun.download_file(s3_file_for("amr_full_results"), local_amr_full_results_path)
```
Comment (cdebourcy) [inline app/models/pipeline_run.rb]: We noticed that loading failed because amr_results was nil. Given that this function only gets called once `check_and_enqueue` detects that the S3 file is present, my only explanation is that `PipelineRun.download_file` randomly failed. My recommendation is: change it to use `download_file_with_retries`, and return early from `db_load_amr_counts` if `amr_results` is still nil. It should then keep doing `db_load_amr_counts` every couple minutes until it succeeds in downloading the file and `amr_results` is not nil. Only then should it move on to the next block.

Comment (cdebourcy) [reply app/models/pipeline_run.rb]: Try that and see how it goes.

Comment (cdebourcy) [reply app/models/pipeline_run.rb]: You'll notice that `db_load_taxon_counts` functions in the way I just described

File: db/migrate/20180726220853_create_amr_counts.rb [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+      t.float :coverage
+      t.float :depth
+      t.bigint :pipeline_run_id
+      t.string :drug_family
+      t.integer :level
+      t.float :drug_gene_coverage
+      t.float :drug_gene_depth
+      t.timestamps
+    end
+    add_index :amr_counts, :allele, unique: true
```
Comment (cdebourcy) [inline db/migrate/20180726220853_create_amr_counts.rb]: This isn't the right unique index to add. This would mean there could only be 1 entry for each allele in the entire table. Instead, it should be restricted to one entry per allele PER PIPELINE RUN. So: `add_index :amr_counts, [:pipeline_run_id, :allele], unique: true`

Comment (Meera-Srinivasan) [reply db/migrate/20180726220853_create_amr_counts.rb]: thanks!

File: app/assets/src/components/AMRView.jsx [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+    accessor: "depth",
+    aggregate: vals => _.sum(vals),
+    filterMethod: (filter, row) => row[filter.id] >= filter.value
+  }
+];
+
+class AMRView extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { data: props.amr ? props.amr : [] };
```
Comment (cdebourcy) [inline app/assets/src/components/AMRView.jsx]: Why is the data put in `state`? Is it a requirement of ReactTable? Normally `state` is used for variables that will change during the session. The AMR data on the other hand seems static.

File: app/models/pipeline_run.rb [hunk 16]
```diff
@@ -278,6 +293,52 @@ def db_load_ercc_counts
     update(total_ercc_reads: ercc_counts_array.map { |entry| entry[:count] }.sum)
   end
 
+  def db_load_amr_counts
+    # test files: "s3://idseq-database/test/AMR/amr_processed_results.csv", "s3://idseq-database/test/AMR/amr_summary_results.csv"
+    # amr_results = PipelineRun.download_file_with_retries("s3://idseq-database/test/AMR/amr_processed_results.csv", local_amr_full_results_path, 3)
+    amr_results = PipelineRun.download_file_with_retries(s3_file_for("amr_full_results"), local_amr_full_results_path, 3)
```
Comment (cdebourcy) [inline app/models/pipeline_run.rb]: Problem is that you have an inconsistency in the name of the output. In `s3_file_for`, you define the path for `amr_full_results` (https://github.com/chanzuckerberg/idseq-web/blob/157a91386f645c5139b65ba655b5d60b59ea8866/app/models/pipeline_run.rb#L420), but in output_states it is `amr_counts` (https://github.com/chanzuckerberg/idseq-web/blob/157a91386f645c5139b65ba655b5d60b59ea8866/app/models/pipeline_run.rb#L183). They need to be the same. Your loading errors stem from the fact that the check for the presence of `s3_file_for("amr_counts")` succeeded but downloading `s3_file_for("amr_full_results")` failed. Once you make the names consistent, it won't try to load until the file is ACTUALLY there

File: app/assets/src/components/AMRView.jsx (hunk 15 above)
Comment (Meera-Srinivasan) [reply app/assets/src/components/AMRView.jsx]: That's true, I can look into it more and get back to you.

File: app/models/pipeline_run.rb (hunk 16 above)
Comment (Meera-Srinivasan) [reply app/models/pipeline_run.rb]: I see, thanks for pointing this out.

File: app/assets/src/components/PipelineSampleReads.jsx [hunk 17]
```diff
@@ -868,12 +869,22 @@ class PipelineSampleReads extends React.Component {
                       Details
                     </a>
                   </li>
+                  <li className="tab">
```
Comment (cdebourcy) [inline app/assets/src/components/PipelineSampleReads.jsx]: check here whether user is admin. If yes, show your <li>. If no, show null.

File: app/helpers/pipeline_runs_helper.rb [hunk 18]
```diff
@@ -4,7 +4,7 @@ module PipelineRunsHelper
   def aegea_batch_submit_command(base_command,
                                  memory = Sample::DEFAULT_MEMORY_IN_MB,
                                  job_queue: nil,
-                                 docker_image: "idseq_dag")
+                                 docker_image: "idseq_dag_meera1")
```
Comment (cdebourcy) [inline app/helpers/pipeline_runs_helper.rb]: Don't change the default here -- call the function with argument "idseq_dag_meera1" in your pipeline_run_stage command

Comment (Meera-Srinivasan) [reply app/helpers/pipeline_runs_helper.rb]: resolved

File: app/models/pipeline_run_stage.rb [hunk 19]
```diff
@@ -212,4 +213,33 @@ def postprocess_command
...
+    file_type = sample.fasta_input? ? 'fasta' : 'fastq'
+    attribute_dict = {
+      fastq1: sample.input_files[0].name,
+      file_type: file_type
+    }
+    attribute_dict[:fastq2] = sample.input_files[1].name if sample.input_files[1]
+    dag_commands = prepare_dag("experimental", attribute_dict)
+
+    # upload_version = "idseq_dag --version | cut -f2 -d ' ' | aws s3 cp  - #{pipeline_run.pipeline_version_file} "
+    batch_command = [install_pipeline(pipeline_run.pipeline_commit), dag_commands].join("; ")
```
[the remark this answers is not in the document]
Comment (Meera-Srinivasan) [reply app/models/pipeline_run_stage.rb]: resolved

File: app/models/pipeline_run_stage.rb [hunk 20]
```diff
@@ -212,4 +213,33 @@ def postprocess_command
...
+
+    # Dispatch job
+    # memory = sample.sample_memory.present? ? sample.sample_memory : Sample::DEFAULT_MEMORY_IN_MB
+    aegea_batch_submit_command(batch_command, job_queue: pipeline_run.sample.job_queue)
```
[the remark this answers is not in the document]
Comment (Meera-Srinivasan) [reply app/models/pipeline_run_stage.rb]: resolved

File: app/controllers/samples_controller.rb [hunk 21]
```diff
@@ -150,7 +150,9 @@ def report_csv
   # GET /samples/1.json
 
   def show
+    @sample = samples_scope.find(params[:id])
```
Comment (yunfangjuan) [inline app/controllers/samples_controller.rb]: I don't think you need this. set_sample should be run by default.

File: app/models/amr_count.rb [hunk 22]
```diff
@@ -0,0 +1,3 @@
+class AmrCount < ApplicationRecord
```
[the remark this answers is not in the document]
Comment (yunfangjuan) [reply app/models/amr_count.rb]: fix this?

File: app/models/amr_count.rb [hunk 23]
```diff
@@ -0,0 +1,14 @@
...
+  belongs_to :pipeline_run
+  def self.viewable(user)
+    if user.admin?
+      all
+    end
```
Comment (yunfangjuan) [inline app/models/amr_count.rb]: so the functions here are not implemented?

Comment (yunfangjuan) [reply app/models/amr_count.rb]: If you are not calling these functions, I don't think you need to implement them here.

File: app/models/sample.rb [hunk 24]
```diff
@@ -292,6 +292,10 @@ def sample_postprocess_s3_path
     "s3://#{SAMPLES_BUCKET_NAME}/#{sample_path}/postprocess"
   end
 
+  def sample_expt_s3_path
```
Comment (yunfangjuan) [inline app/models/sample.rb]: add some comments so people understand expt means experimental.

File: app/models/sample.rb [hunk 25]
```diff
@@ -466,7 +470,7 @@ def kickoff_pipeline
     # The subsample field of "sample" is currently used as a simple flag (UI checkbox),
     # but was made an integer type in case we want to allow users to enter the desired number
     # of reads to susbample to in the future
-    pr.pipeline_branch = pipeline_branch.blank? ? "master" : pipeline_branch
+    pr.pipeline_branch = pipeline_branch.blank? ? "meera-add-amr" : pipeline_branch
```
Comment (yunfangjuan) [inline app/models/sample.rb]: this is dangerous. please change back to master.

File: app/models/pipeline_run_stage.rb [hunk 26]
```diff
@@ -212,4 +213,33 @@ def postprocess_command
...
+  end
+
+  def assembly_command
```
[the remark this answers is not in the document]
Comment (Meera-Srinivasan) [reply app/models/pipeline_run_stage.rb]: resolved

File: app/controllers/samples_controller.rb (hunk 21 above)
Comment (Meera-Srinivasan) [reply app/controllers/samples_controller.rb]: Resolved

File: app/models/sample.rb (hunk 25 above)
Comment (Meera-Srinivasan) [reply app/models/sample.rb]: resolved

File: app/models/sample.rb (hunk 24 above)
Comment (Meera-Srinivasan) [reply app/models/sample.rb]: resolved

File: app/models/amr_count.rb (hunk 22 above)
[the remark this answers is not in the document]
Comment (Meera-Srinivasan) [reply app/models/amr_count.rb]: resolved

File: app/assets/src/components/PipelineSampleReads.jsx [hunk 27]
```diff
@@ -30,6 +32,7 @@ class PipelineSampleReads extends React.Component {
     this.reportDetails = props.reportDetails;
     this.pipelineRunRetriable = props.pipelineRunRetriable;
     this.pipelineVersions = props.pipeline_versions;
+    this.pipelineVersion = props.pipeline_version;
```
Comment (cdebourcy) [inline app/assets/src/components/PipelineSampleReads.jsx]: remove this?

File: app/controllers/samples_controller.rb [hunk 28]
```diff
@@ -189,6 +196,10 @@ def show
     end
   end
 
+  def show_amr_counts
```
Comment (cdebourcy) [inline app/controllers/samples_controller.rb]: Remove this?

File: app/models/pipeline_run.rb [hunk 29]
```diff
@@ -28,9 +30,14 @@ class PipelineRun < ApplicationRecord
   PIPELINE_VERSION_FILE = "pipeline_version.txt".freeze
   STATS_JSON_NAME = "stats.json".freeze
   ERCC_OUTPUT_NAME = 'reads_per_gene.star.tab'.freeze
+  AMR_DRUG_SUMMARY_RESULTS = 'amr_summary_results.csv'.freeze
+  AMR_FULL_RESULTS_NAME = 'amr_processed_results.csv'.freeze
   TAXID_BYTERANGE_JSON_NAME = 'taxid_locations_combined.json'.freeze
   ASSEMBLY_STATUSFILE = 'job-complete'.freeze
   LOCAL_JSON_PATH = '/app/tmp/results_json'.freeze
+  LOCAL_AMR_FULL_RESULTS_PATH = '/app/tmp/amr_full_results'.freeze
+  # Note -- not currently being used, but leaving here for reference
```
Comment (cdebourcy) [inline app/models/pipeline_run.rb]: If not used, please remove it

File: app/models/pipeline_run.rb [hunk 30]
```diff
@@ -285,6 +300,41 @@ def db_load_ercc_counts
...
+                              depth:  amr_result_fields[3],
+                              drug_family: amr_result_fields[12] }
+      end
+      update(amr_counts_attributes: amr_counts_array)
+    end
+  end
+
+  # Can remove as aggregation is being done on the UI side,
+  # but leaving as useful function for possible future changes (e.g. if UI table)
+  def summarize_amr_drug_family_counts
```
Comment (cdebourcy) [inline app/models/pipeline_run.rb]: Since it is not being used, please remove it

File: app/assets/src/components/AMRView.jsx (hunk 15 above)
Comment (tfrcarvalho) [reply app/assets/src/components/AMRView.jsx]: As recommended by @cdebourcy, you should move the data out of state as you are not using it to influence any local elements.
This could even be a functional component.

File: app/views/samples/show_amr_counts.html.erb [hunk 31]
```diff
@@ -0,0 +1,7 @@
+<div id="AMRView_page">
```
Comment (cdebourcy) [inline app/views/samples/show_amr_counts.html.erb]: Since we have the tab now, we don't need this page anymore. Please remove it

File: config/routes.rb [hunk 32]
```diff
@@ -12,6 +12,7 @@
     put :retry_pipeline, on: :member
     get :all, on: :collection
     get :pipeline_runs, on: :member
+    get :show_amr_counts, on: :member
```
Comment (cdebourcy) [inline config/routes.rb]: remove this

File: app/controllers/samples_controller.rb [hunk 33]
```diff
@@ -14,7 +14,7 @@ class SamplesController < ApplicationController
   ##########################################
   skip_before_action :verify_authenticity_token, only: [:create, :update]
 
-  READ_ACTIONS = [:show, :report_info, :search_list, :report_csv, :assembly, :show_taxid_fasta, :nonhost_fasta, :unidentified_fasta, :results_folder, :fastqs_folder, :show_taxid_alignment, :show_taxid_alignment_viz].freeze
+  READ_ACTIONS = [:show, :report_info, :search_list, :report_csv, :assembly, :show_taxid_fasta, :nonhost_fasta, :unidentified_fasta, :results_folder, :fastqs_folder, :show_taxid_alignment, :show_taxid_alignment_viz, :show_amr_counts].freeze
```
Comment (cdebourcy) [inline app/controllers/samples_controller.rb]: remove this change

File: app/models/pipeline_run_stage.rb [hunk 34]
```diff
@@ -212,4 +213,23 @@ def postprocess_command
...
+    sample = pipeline_run.sample
+    file_type = sample.fasta_input? ? 'fasta' : 'fastq'
+    attribute_dict = {
+      fastq1: sample.input_files[0].name,
+      file_type: file_type
+    }
+    attribute_dict[:fastq2] = sample.input_files[1].name if sample.input_files[1]
+    dag_commands = prepare_dag("experimental", attribute_dict)
+
+    # upload_version = "idseq_dag --version | cut -f2 -d ' ' | aws s3 cp  - #{pipeline_run.pipeline_version_file} "
```
Comment (cdebourcy) [inline app/models/pipeline_run_stage.rb]: remove this commented line

File: app/assets/src/components/PipelineSampleReads.jsx [hunk 35]
```diff
@@ -834,6 +836,21 @@ class PipelineSampleReads extends React.Component {
           </div>
         </div>
       );
+
+    let show_amr = this.amr != null;
+    let amr_tab = show_amr ? (
+      <li className="tab">
+        <a href="#amr" className="">
+          Antimicrobial Resistance Report
```
Comment (cdebourcy) [inline app/assets/src/components/PipelineSampleReads.jsx]: This is a bit wordy. Change it to just "Antimicrobial Resistance"

File: app/assets/src/components/AMRView.jsx (hunk 15 above)
Comment (cdebourcy) [reply app/assets/src/components/AMRView.jsx]: So for now, just literally change `this.state` to `this.data` or whatever

Comment (Meera-Srinivasan) [reply app/assets/src/components/AMRView.jsx]: Sounds good, resolved

File: app/assets/src/components/PipelineSampleReads.jsx (hunk 27 above)
Comment (Meera-Srinivasan) [reply app/assets/src/components/PipelineSampleReads.jsx]: resolved

File: app/controllers/samples_controller.rb (hunk 33 above)
Comment (Meera-Srinivasan) [reply app/controllers/samples_controller.rb]: done

File: config/routes.rb (hunk 32 above)
Comment (Meera-Srinivasan) [reply config/routes.rb]: done

File: app/controllers/samples_controller.rb (hunk 28 above)
Comment (Meera-Srinivasan) [reply app/controllers/samples_controller.rb]: done

File: app/models/pipeline_run.rb (hunk 30 above)
Comment (Meera-Srinivasan) [reply app/models/pipeline_run.rb]: done

File: app/models/pipeline_run.rb (hunk 29 above)
Comment (Meera-Srinivasan) [reply app/models/pipeline_run.rb]: done

File: app/views/samples/show_amr_counts.html.erb (hunk 31 above)
Comment (Meera-Srinivasan) [reply app/views/samples/show_amr_counts.html.erb]: done

File: app/models/pipeline_run_stage.rb (hunk 34 above)
Comment (Meera-Srinivasan) [reply app/models/pipeline_run_stage.rb]: done

File: app/assets/src/components/PipelineSampleReads.jsx (hunk 35 above)
Comment (Meera-Srinivasan) [reply app/assets/src/components/PipelineSampleReads.jsx]: done
